### ISSUE TITLE
Generate enum reverse lookup

### ISSUE THREAD

**Issue by keynmol** [indoorvivants/sn-bindgen#134]
Take this beautiful, voluptuous enum:

```scala
  opaque type ConnStatusType = CUnsignedInt
  object ConnStatusType extends CEnumU[ConnStatusType]:
    given _tag: Tag[ConnStatusType] = Tag.UInt
    inline def define(inline a: Long): ConnStatusType = a.toUInt
    val CONNECTION_OK = define(0)
    val CONNECTION_BAD = define(1)
    val CONNECTION_STARTED = define(2)
    val CONNECTION_MADE = define(3)
    val CONNECTION_AWAITING_RESPONSE = define(4)
    val CONNECTION_AUTH_OK = define(5)
    val CONNECTION_SETENV = define(6)
    val CONNECTION_SSL_STARTUP = define(7)
    val CONNECTION_NEEDED = define(8)
    val CONNECTION_CHECK_WRITABLE = define(9)
    val CONNECTION_CONSUME = define(10)
    val CONNECTION_GSS_STARTUP = define(11)
    val CONNECTION_CHECK_TARGET = define(12)
    val CONNECTION_CHECK_STANDBY = define(13)
    extension (a: ConnStatusType)
      inline def &(b: ConnStatusType): ConnStatusType = a & b
      inline def |(b: ConnStatusType): ConnStatusType = a | b
      inline def is(b: ConnStatusType): Boolean = (a & b) == b
```

When you get a status type, you can compare it to pre-defined values.
But what if it's an unknown status type? 
We should generate a

```scala
inline def getName(inline value: ConnStatusType): Option[String] =
   value match 
     case  CONNECTION_CONSUME => Some("CONNECTION_CONSUME")
     ... 
     case _ => None

```

**Comment by cheleb** [comment]
https://github.com/indoorvivants/sn-bindgen/pull/138

**Comment by keynmol** [comment]
Closed in #138 (target release is 0.0.15)
